### PR TITLE
feat(h13): add Blurhash to Artworks 

### PR DESCRIPTION
**Pull request by MrSltun**
# [Last UPDATE to description]: Monday, March 4th 2023
## Reviewable 🎉  
This PR resolves [ONYX-674] 

### Description

This PR adds Blurhashes as a placeholder to our artwork images component. To achieve that, a few changes needed to be made:

#### We are currently using 5 different `Image` components
- OpaqueImageView
- OpaqueImageView2
- React Native Image
- Palette mobile Image
- FastImage

I moved everything to be relying on `@artsy/palette-mobile`, **unless**, it's a local file, where I didn't feel it makes sense to migrate them, yet. But because this is a big change, I didn't remove the files yet, until we're sure that the new component is stable

> [!NOTE]  
> The above work is hidden behind a feature flag: "use palette image".

#### Support blurhashes on palette-mobile

See: https://github.com/artsy/palette-mobile/pull/173

#### Add blurhash to artwork image

> [!NOTE]  
> The above work is hidden behind a feature flag: "Show blurhash".

| iOS |
| --- |
| <Video src="https://github.com/artsy/eigen/assets/11945712/92ac8cdf-bcec-4549-9360-a7adc58ab4fd" /> |

| Android |
| --- |
| <Video src="https://github.com/artsy/eigen/assets/11945712/6ae30fd1-7819-40ee-b6d4-6871aaa5493c" /> |

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Introduce BlurHash to Eigen - mrsltun

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-674]: https://artsyproduct.atlassian.net/browse/ONYX-674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b7b7383805e1c8a7abd1f010ae363b227c956296
</p>

File: src/app/Components/OpaqueImageView/OpaqueImageView.tsx [hunk 1]
```diff
@@ -1,186 +1,3 @@
...
-        style={[style, backgroundColorStyle] as any}
-        {...remainderProps}
-        source={{ uri: Image.resolveAssetSource({ uri: remainderProps.imageURL! }).uri }}
-      />
-    )
-  }
-}
-
-const NativeOpaqueImageView = requireNativeComponent("AROpaqueImageView") as typeof View
+export default OpaqueImageView
```
Comment (MounirDhahri) [inline src/app/Components/OpaqueImageView/OpaqueImageView.tsx:3]: @gkartalis @brainbicycle  do we need to keep the old component? I think we can already move towards `OpaqueImageView2` and if the ff is enabled `Image` from `palette

File: src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx [hunk 2]
```diff
@@ -97,26 +107,26 @@ export const OpaqueImageView: React.FC<Props> = ({ aspectRatio, ...props }) => {
...
       return [layoutHeight * aspectRatio!, layoutHeight]
     }
     return [layoutWidth, layoutHeight]
   }, [props.height, props.width, style.width, style.height, aspectRatio, layoutHeight, layoutWidth])
 
-  useEffect(() => {
-    const [fWidth, fHeight] = getActualDimensions()
-    setFIHeight(fHeight)
-    setFIWidth(fWidth)
-  }, [getActualDimensions])
```
Comment (MounirDhahri) [inline src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx:119]: this was leading to an unnecessary re-rerender, I changed it in favor of just getting the value directly. I didn't feel the need to memorize it, let me know what you think

File: src/app/Scenes/Inbox/Components/ActiveBids/ActiveBid.tsx [hunk 3]
```diff
@@ -113,47 +112,41 @@ class ActiveBid extends React.Component<Props, State> {
...
+    const blurhash = bid?.sale_artwork?.artwork?.image?.blurhash
+
+    const lotNumber = bid?.sale_artwork?.lot_label || ""
+    const artistName = bid?.sale_artwork?.artwork?.artist_names || ""
 
     const headline = `Lot ${lotNumber} · ${artistName} `
 
     const isInOpenLiveAuction = this.props.bid.sale && this.props.bid.sale.is_live_open
-    // @ts-expect-error STRICTNESS_MIGRATION --- 🚨 Unsafe legacy code 🚨 Please delete this and fix any type errors if you have time 🙏
-    const bidderPositions = bid.sale_artwork.counts.bidder_positions
```
Comment (MounirDhahri) [inline src/app/Scenes/Inbox/Components/ActiveBids/ActiveBid.tsx:142]: this felt good 😄

**Review by MounirDhahri — COMMENTED** [review]
Next steps from here would be:
- Remove `OpaqueImageView` and use `Image` everywhere
- Look into making sure **ALL** images have a blurhash, many images didn't have any
- QA this together and release the feature if everything looks alright

File: src/app/Components/OpaqueImageView/OpaqueImageView.tsx (hunk 1 above)
Comment (gkartalis) [reply src/app/Components/OpaqueImageView/OpaqueImageView.tsx:3]: no need to keep the old one

Comment (gkartalis) [reply src/app/Components/OpaqueImageView/OpaqueImageView.tsx:3]: we just need to make sure that the image component from palette actually works in all the places

File: src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx (hunk 2 above)
Comment (gkartalis) [reply src/app/Components/OpaqueImageView2/OpaqueImageView2.tsx:119]: 👋

**Comment by MounirDhahri** [comment]
We spoke about this today at the mobile office hours and we are good to go

**Pull request merged**